src/store/actions/login.js

```diff
@@ -0,0 +1,22 @@
+import axios from 'axios';
+import { LOGIN } from './types';
+import { API_BASE_URL } from '../../settings';
+import { SUCCESS, FAILURE } from '../../constants';
+
+export const login = (status, payload) => ({
+  type: LOGIN[status],
+  payload,
+});
+
+export const loginAPI = user => async (dispatch) => {
+  try {
+    const response = await axios({
+      url: `${API_BASE_URL}/users/login`,
```

Comment:
I would strongly suggest the implementation of a middleware to handle your API requests. You have a lot of repeated code:
1. prepending `API_BASE_URL` to your endpoint
2. triggering `dispatch` on the success and failure states and passing the response data to the redux store

This does not allow you to scale or refactor any of this logic quickly across you application.

You can handle this in a separate PR. But it should be handled as soon as possible as it is impacting other design decisions in your application. Perhaps you could even pause this PR until the middleware refactor is complete.

Reply:
I will open  a PR after this to handle it as to not block this PR. i have created a ticket https://github.com/AntonioLok/online-shopping-reactJS-Frontend/projects/1 and its my next to-do item